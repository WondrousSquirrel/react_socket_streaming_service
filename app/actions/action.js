const { RECIEVE_DATA } = require('./types');

export const recieveDataAction = data => ({
    type: RECIEVE_DATA,
    payload: data
});
