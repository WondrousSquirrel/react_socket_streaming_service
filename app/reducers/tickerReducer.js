const initialState = {};

export default (state = initialState, {type, payload}) => {
    switch(type) {
        case 'RECIEVE_DATA':
            return {
                ...state, data: payload
            };
        default:
            return state;
    }
};
