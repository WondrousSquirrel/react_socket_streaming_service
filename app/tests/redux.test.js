import { recieveDataAction } from '../actions/action';
import { RECIEVE_DATA } from '../actions/types';
import tickerReducer from '../reducers/tickerReducer';


describe('actions', () => {
    it('should create an action to add a todo', () => {
        const data = 'test case';
        /*
        const expectedAction = {
            type: 'RECIEVE_DATA',
            payload: data
        };
        */
        expect(recieveDataAction(data)).toMatchSnapshot();
    });
});

describe('reducers', () => {
    describe('Initial State', () => {
        test('always pass', () => {
            const action = {type: 'EMPTY_ACTION'};
            const initialState = { price: 0 };
            expect(tickerReducer(initialState, action)).toMatchSnapshot();
        });
    });
    describe('recieve Ddata', () => {
        test('returns the correct state', () => {
            const action = { type: RECIEVE_DATA, payload: 1 };
            // const expectedState = { data: 1 };
            expect(tickerReducer(undefined, action)).toMatchSnapshot();
        });
    });
});
