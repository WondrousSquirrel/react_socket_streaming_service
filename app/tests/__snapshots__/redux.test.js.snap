// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should create an action to add a todo 1`] = `
Object {
  "payload": "test case",
  "type": "RECIEVE_DATA",
}
`;

exports[`reducers Initial State always pass 1`] = `
Object {
  "price": 0,
}
`;

exports[`reducers recieve Ddata returns the correct state 1`] = `
Object {
  "data": 1,
}
`;
